src/map/pc.c

```diff
@@ -5277,7 +5277,7 @@ enum e_setpos pc_setpos(struct map_session_data* sd, unsigned short mapindex, in
 		sd->state.pmap = sd->bl.m;
 		if (sd->sc.count) { // Cancel some map related stuff.
 			if (sd->sc.data[SC_JAILED])
-				return 1; //You may not get out!
+				return SETPOS_MAPINDEX; //You may not get out!
```

Comment:
If we specifically check somewhere for when a player is jailed and return a unique jailed message, then yes. If not we shouldn't really need any unique value for it.